File: Sowndariya_Assessment24/Pages/loginPage.py
from selenium.webdriver.common.by import By

class LoginPage:

    def __init__(self, driver):
        self.driver = driver

    def my_account(self):
        return self.driver.find_element(By.XPATH,"//span[text()='My Account']")

    def login_link(self):
        return self.driver.find_element(By.LINK_TEXT,"Login")

    def email(self):
        return self.driver.find_element(By.ID,"input-email")

    def password(self):
        return self.driver.find_element(By.ID,"input-password")

    def login_btn(self):
        return self.driver.find_element(By.XPATH,"//input[@value='Login']")

    def error(self):
        return self.driver.find_element(By.CSS_SELECTOR,".alert-danger")

    def account_name(self):
        return self.driver.find_element(
            By.XPATH,
            "//h2[text()='My Account']"
        )

File: Sowndariya_Assesssment03/TechnicalAssessment03/src/test/java/com/pages/LoginPage.java
package com.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage {

    WebDriver driver;

    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void clickMyAccount() {
        driver.findElement(By.linkText("My Account")).click();
    }

    public void clickLogin() {
        driver.findElement(By.linkText("Login")).click();
    }

    public void enterEmail(String email) {
        driver.findElement(By.id("input-email")).sendKeys(email);
    }

    public void enterPassword(String password) {
        driver.findElement(By.id("input-password")).sendKeys(password);
    }

    public void clickLoginButton() {
        driver.findElement(By.xpath("//input[@value='Login']")).click();
    }
}